pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -205,9 +208,9 @@ func (pl *VolumeBinding) Filter(ctx context.Context, cs *framework.CycleState, p
 		return status
 	}
 
-	cs.Lock()
+	state.Mutex.Lock()
```

Comment:
```suggestion
    // multiple goroutines call `Filter` on different nodes simultaneously and the `CycleState` may be duplicated, so we must use a local lock here
	state.Lock()
```

I would suggest adding the comment here explaining why we can't rely on `framework.CycleState`'s lock.